src/node/blockstorage.cpp

```diff
@@ -265,13 +265,26 @@ CBlockIndex* BlockManager::InsertBlockIndex(const uint256& hash)
     return pindex;
 }
 
-bool BlockManager::LoadBlockIndex()
+bool BlockManager::LoadBlockIndex(const std::optional<uint256>& snapshot_blockhash)
 {
     if (!m_block_tree_db->LoadBlockIndexGuts(
             GetConsensus(), [this](const uint256& hash) EXCLUSIVE_LOCKS_REQUIRED(cs_main) { return this->InsertBlockIndex(hash); }, m_interrupt)) {
         return false;
     }
 
+    int snapshot_height = -1;
+    if (snapshot_blockhash) {
+        const AssumeutxoData au_data = *Assert(GetParams().AssumeutxoForBlockhash(*snapshot_blockhash));
+        snapshot_height = au_data.height;
+        CBlockIndex* base{LookupBlockIndex(*snapshot_blockhash)};
+
+        // Since nChainTx (responsible for estiamted progress) isn't persisted
```

Comment:
In commit "validation: populate nChainTx value for assumedvalid chainstates" (e3cae85738c8bf4116547af8d868c6c01577b80f)

estimated